app/ante/eth.go

```diff
@@ -331,14 +331,15 @@ func (ctd CanTransferDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 	}
 
 	ethCfg := config.EthereumConfig(ctd.evmKeeper.ChainID())
+	signer := ethtypes.MakeSigner(ethCfg, big.NewInt(ctx.BlockHeight()))
 
 	for _, msg := range tx.GetMsgs() {
 		msgEthTx, ok := msg.(*evmtypes.MsgEthereumTx)
 		if !ok {
 			return ctx, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "invalid transaction type: %T", msg)
 		}
 
-		coreMsg, err := msgEthTx.AsMessage()
+		coreMsg, err := msgEthTx.AsMessage(signer)
```

Comment:
Nice change, 

Please check also this issue #110

Basically all the AnteHandle (not only CanTransferDecorator) should follow the same logic. One possible solution is also the set the field 'from' automatically at higher level if there is too much refactoring